app/javascript/components/admin/UserModal.js

```diff
@@ -71,69 +78,108 @@ class UserModal extends React.Component {
             <Form.Group>
-              <Form.Label htmlFor="role-select">Role</Form.Label>
+              <Form.Label className="input-label" htmlFor="role-select">
+                Role
+              </Form.Label>
               <Select
                 inputId="role-select"
                 name="role"
-                defaultValue={
-                  this.props.initialUserData.role_title
-                    ? { label: this.props.initialUserData.role_title, value: this.props.initialUserData.jurisdiction_path }
+                value={
+                  this.state.roleTitle
+                    ? { label: this.state.roleTitle, value: this.state.roleTitle }
                     : { label: this.props.roles[0], value: this.props.roles[0] }
                 }
                 options={this.props.roles.map(role => {
                   return { label: role, value: role };
                 })}
-                onChange={this.handleRoleChange}
+                onChange={data => this.setState({ roleTitle: data.value })}
                 placeholder=""
                 styles={cursorPointerStyle}
                 theme={bootstrapSelectTheme}
               />
             </Form.Group>
             {this.props.type === 'edit' && (
               <Form.Group>
-                <Form.Label>Status</Form.Label>
-                <Form.Check
-                  id="status-input"
-                  name="status"
-                  type="switch"
-                  checked={this.state.isLocked}
-                  label={this.state.isLocked ? 'Locked' : 'Unlocked'}
-                  onChange={this.handleLockedStatusChange}
-                />
+                <Form.Label className="input-label">System Access</Form.Label>
+                <div style={{ display: 'flex', alignItems: 'center' }}>
+                  <Form.Check
+                    id="system-access-input"
+                    name="isLocked"
+                    type="switch"
+                    checked={this.state.isLocked}
+                    label={this.state.isLocked ? 'Locked' : 'Unlocked'}
+                    onChange={this.handleChange}
+                  />
+                  {this.state.lockReason === 'Auto-locked by the System' && this.state.isLocked && (
+                    <div className="locked-warning-text ml-2">
+                      <FontAwesomeIcon className="mr-1" icon={faExclamationCircle} />
+                      <span>{this.state.autoLockReason}</span>
+                    </div>
+                  )}
+                </div>
+              </Form.Group>
+            )}
+            {this.props.type === 'edit' && (
+              <Form.Group>
+                <Form.Label className="input-label" htmlFor="status-select">
+                  Status
+                </Form.Label>
+                <span data-for="disabled-status-select" data-tip="">
+                  <Select
+                    inputId="status-select"
+                    id="status"
```

Comment:
not sure this is needed because the onChange uses name

Reply:
Same commit as above: [d2f14f7ed8eaadf3fd01b5e4a9174f88bc436f87](https://github.com/SaraAlert/SaraAlert/pull/1180/commits/d2f14f7ed8eaadf3fd01b5e4a9174f88bc436f87)